lib/compress.js

```diff
@@ -4895,6 +4896,18 @@ merge(Compressor.prototype, {
                 });
             }
         }
+        // ["p"]:1 ---> p:1
+        // [42]:1 ---> 42:1
+        if (compressor.option("computed_props")
+            && self.key instanceof AST_Constant // save a comparison in the typical case
+            && (
+                // whitelist acceptable props as AST_Constants are not all constant
+                self.key instanceof AST_String
+                || self.key instanceof AST_Number
+            )) {
+            self.key = self.key.value;
+            return self;
```

Comment:
I think we can forego this `return` statement as this seems safe to chain with future optimisation rules.

Reply:
I also had that thought. I'll move it above other the OPT transform and not return with a suitable comment.